### DISCUSSION TITLE
Constraint / Validator multiple atPath

### DISCUSSION THREAD

**Discussion by devbysb** [symfony/symfony#47662] [Q&A] [answered]
### Description

Hi 
Is it possible to add multiple atPath() or pass on array of fields or add a function addAtPath() ... to show error on 2 fields for example.

I'm using SF5.4

Thank you

### Example

`        
       $phone = $value->phone ?? '';
        $mail = $value->mail ?? '';

        if ($phone !== '' || $mail !== '') {
            return;
        }

        $this->context->buildViolation($constraint->message)
            ->atPath('phone')
            ->atPath('mail') // the last is chosen and I want to show on 2 fields
            ->addViolation();`

**Comment by stof** [comment, marked as answer]
For that, you would have to clone your ConstraintViolationBuilder to add violation multiple times with different atPath. Your current code adds a single violation, not 2.

**Reply by devbysb** [reply]
Sorry I have not precise, but the constraint is apply on class (getTagets()) not on property
How you can clone ConstraintViolationBuilder ?

**Reply by stof** [reply]
```php
$constraintViolationBuilder = $this->context->buildViolation($constraint->message);

$secondViolationBuilder = clone $constraintViolationBuilder;
```

And now, you have 2 builders sharing the initial config done on them, that you can each finish configurating and then call `addViolation()`.

However, if the only shared config is the message, it might be easier to call `buildViolation` twice instead of cloning.

**Reply by devbysb** [reply]
Thank you :)
